quarkus-opentelemetry-exporter-gcp/deployment/src/test/java/io/quarkiverse/opentelemetry/exporter/gcp/deployment/GcpExporterEnabledTest.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkiverse.opentelemetry.exporter.gcp.deployment;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.quarkiverse.opentelemetry.exporter.gcp.runtime.LateBoundBatchSpanProcessor;
+import io.quarkiverse.opentelemetry.exporter.gcp.runtime.LateBoundSimpleSpanProcessor;
+import io.quarkus.test.QuarkusUnitTest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+public class GcpExporterEnabledTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withEmptyApplication()
```

Comment:
Hello @brunobat,

At the moment this test fails. I need a bit of help to solve this.

Summary:
- I initialize GCP's `TraceExporter` in the `GcpRecord` class. Initialization depends if it's CloudRun environment or not.
- Every time Google's `TraceExporter` is initialized it expects GCP's ProjectID and Credentials present.
- I've reviewed `opentelemetry-operations-java` github repo and found a way to do E2E test but I need to override/replace `TraceExporter` configuration.

What would be the best way to do it? Should I create different `Recorder` for tests?

Reply:
Let me checkout the code and see what I can do.

Reply:
Thanks! I can join Quarkus Zulip Chat instance so we can discuss.

Reply:
Sure, ping me on Zullip